File: src/test/java/io/github/jho951/assignment/job/service/RequestHashServiceTests.java
package io.github.jho951.assignment.job.service;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class RequestHashServiceTests {

    private final RequestHashService requestHashService = new RequestHashService();

    @Test
    void shouldReturnStableHashForSameImageUrl() {
        String imageUrl = "https://example.com/input.png";

        String first = requestHashService.hashImageUrl(imageUrl);
        String second = requestHashService.hashImageUrl(imageUrl);

        assertThat(first).isEqualTo(second);
        assertThat(first).hasSize(64);
    }

    @Test
    void shouldReturnDifferentHashesForDifferentImageUrls() {
        String first = requestHashService.hashImageUrl("https://example.com/input-1.png");
        String second = requestHashService.hashImageUrl("https://example.com/input-2.png");

        assertThat(first).isNotEqualTo(second);
    }
}

File: src/test/kotlin/io/github/jho951/assignment/job/service/RequestHashServiceTests.kt
package io.github.jho951.assignment.job.service

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class RequestHashServiceTests {

    private val requestHashService = RequestHashService()

    @Test
    fun shouldReturnStableHashForSameImageUrl() {
        val imageUrl = "https://example.com/input.png"

        val first = requestHashService.hashImageUrl(imageUrl)
        val second = requestHashService.hashImageUrl(imageUrl)

        assertThat(first).isEqualTo(second)
        assertThat(first).hasSize(64)
    }

    @Test
    fun shouldReturnDifferentHashesForDifferentImageUrls() {
        val first = requestHashService.hashImageUrl("https://example.com/input-1.png")
        val second = requestHashService.hashImageUrl("https://example.com/input-2.png")

        assertThat(first).isNotEqualTo(second)
    }
}
